cloud/softlayer/sl.py

```diff
@@ -0,0 +1,430 @@
+    required: false
+    default: 'present'
+  quantity:
+    description:
+      - Quantity of virtual instances to provision
+    required: false
+    default: 0
+  increment:
+    description:
+      - Flag used to determine if a incremental value should be appended to hostname
+    required: false
+    default: false
+  increment_start:
+    description:
+      - Number to start increment from
+    required: false
+    default: 1
+  increment_delimiter:
+    description:
+      - Character to use as delimiter for increment
+    required: false
+    default: ''
+  wait:
+    description:
+      - Flag used to wait for active status before returning
+    required: false
+    default: true
+  wait_timeout:
+    description:
+      - time in seconds before wait returns
+    required: false
+    default: 600
+
+requirements:
+    - "python >= 2.6"
+author: "Matt Colton (@mcltn)"
+'''
+
+EXAMPLES = '''
+- name: Build instance
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Build instance request
+    local_action:
+      module: softlayer
+      hostname: instance-1
+      domain: anydomain.com
+      datacenter: dal09
+      tags: ansible-module-test
+      hourly: True
+      private: False
+      dedicated: False
+      local_disk: True
+      cpus: 1
+      memory: 1024
+      disks: [25]
+      os_code: UBUNTU_LATEST
+      wait: False
+
+- name: Build additional instances
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Build instances request
+    local_action:
+      module: softlayer
+      quantity: 2
+      increment: True
+      increment_start: 2
+      increment_delimiter: '-'
+      hostname: instance
+      domain: anydomain.com
+      datacenter: dal09
+      tags: ['ansible-module-test', 'ansible-module-test-slaves']
+      hourly: True
+      private: False
+      dedicated: False
+      local_disk: True
+      cpus: 1
+      memory: 1024
+      disks: [25,100]
+      os_code: UBUNTU_LATEST
+      image_id:
+      private_vlan:
+      ssh_keys: []
+      post_uri:
+
+- name: Cancel instances
+  hosts: localhost
+  gather_facts: False
+  tasks:
+  - name: Cancel by tag
+    local_action:
+      module: softlayer
+      state: absent
+      tags: ansible-module-test
+'''
+
+# TODO: Disabled RETURN as it is breaking the build for docs. Needs to be fixed.
+RETURN = '''# '''
+
+import time
+
+STATES = ['present', 'absent']
+DATACENTERS = ['ams01','ams03','dal01','dal05','dal06','dal09','fra02','hkg02','hou02','lon2','mel01','mex01','mil01','mon01','par01','sjc01','sjc03','sao01','sea01','sng01','syd01','tok02','tor01','wdc01','wdc04']
+CPU_SIZES = [1,2,4,8,16]
+MEMORY_SIZES = [1024,2048,4096,6144,8192,12288,16384,32768,49152,65536]
+INITIALDISK_SIZES = [25,100]
+LOCALDISK_SIZES = [25,100,150,200,300]
+SANDISK_SIZES = [10,20,25,30,40,50,75,100,125,150,175,200,250,300,350,400,500,750,1000,1500,2000]
+NIC_SPEEDS = [10,100,1000]
+
```

Comment:
These could possibly be pulled via Api calls, but are also pretty static. Only one that may change in near future would be data centers.